docs/src/docs/asciidoc/maven-plugin.adoc

```diff
@@ -114,10 +114,118 @@ Build Configuration]. It is also possible to customize the plugin within a
    image name is not supplied, the artifact ID of the project will be used by default.
 `<buildArgs>`::
    If you want to pass additional arguments to the native image builder, use `<buildArgs>`
-   in the configuration of the plugin.
+   in the configuration of the plugin:
+[source,xml]
+----
+<buildArgs>
+    <arg>--argument</arg>
+</buildArgs>
+----
 `<skipNativeBuild>`::
-   To skip generation of the native image, supply
-   `<skipNativeBuild>true</skipNativeBuild>` in the configuration of the plugin.
+   To skip generation of the native image, supply the following in the configuration of the plugin:
+[source,xml]
+----
+<skipNativeBuild>true</skipNativeBuild>
+----
+`<skipNativeTests>`::
+   To skip generation and execution of the native image compiled tests, supply the following in the configuration of the plugin:
+[source,xml]
+----
+<skipNativeTests>true</skipNativeTests>
+----
+`<debug>`::
+   If you want to enable generation of debugging information supply the following in the configuration of the plugin:
+[source,xml]
+----
+<debug>true</debug>
+----
+`<verbose>`::
+   If you want to enable verbose output during native-image building supply the following in the configuration of the plugin:
+[source,xml]
+----
+<verbose>true</verbose>
+----
+`<sharedLibrary>`::
+   If you want to build image as a shared library supply the following in the configuration of the plugin:
+[source,xml]
+----
+<sharedLibrary>true</sharedLibrary>
+----
+`<useArgFile>`::
+   If you want to use argument file for native-image building supply the following in the configuration of the plugin:
+[source,xml]
+----
+<useArgFile>true</useArgFile>
+----
+`<quickBuild>`::
+   If you want to build the image using https://blogs.oracle.com/java/post/graalvm-enterprise-221--faster-smarter-leaner[quick build mode], supply the following in the configuration of the plugin:
+[source,xml]
+----
+<quickBuild>true</quickBuild>
```

Comment:
So in addition to this, what if we allowed users to set this mode also from an environment variable? 

For example, the Micronaut Maven Plugin has quite a few integration tests that involve creating a native image. Instead of going POM by POM adding this config, I wish I could simply pass an environment variable from the GitHub Actions workflow.

Reply:
Well you could pass `-DquickBuild` and that should just work. 
I don't know if there is some nice way to also consume properties from env, but I could add a explicit check for `System.getenv("quickBuild")` as well?

Reply:
> Well you could pass `-DquickBuild` and that should just work.

That will involve changing a few places too.

> I don't know if there is some nice way to also consume properties from env, but I could add a explicit check for `System.getenv("quickBuild")` as well?

For plugin authors, there is no other way than `System.getenv()` as far as I know.

I would, however, look for a more specific environment variable name, e.g. `GRAALVM_QUIICK_BUILD`

Reply:
Implementation is great! The only thing left is documenting it here